build.gradle

```diff
@@ -22,6 +22,17 @@ dependencies {
 
 mainClassName = 'com.gradle.exportapi.Application'
 
+task addAuthorization {
+    description 'Adds authorization to the system properties to be available at runtime if configured'
+    onlyIf {
+        System.getProperty("basic_auth") == null && System.getProperty("password") != null
+    }
+    doLast {
+        def username = System.getProperty("username")
+        tasks.run.systemProperties.put("basic_auth", "$username:${System.getProperty("password")}".bytes.encodeBase64().toString())
```

Comment:
Your logic is perfectly sound.  But in this case, as it's the 'run' task, it's the final stage in the application.  You would never want the `run` task to be cached, or up-to-date, etc.

We could also solve this problem by injecting into the System.properties and then the run task picks it up automatically, but I was trying to reduce the scope of where we put these credentials.  This might be something I revisit in the future to clean up.